spec/models/dossier_notification_spec.rb

```diff
@@ -1,5 +1,60 @@
-require 'rails_helper'
+# frozen_string_literal: true
 
 RSpec.describe DossierNotification, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe '.to_display' do
+    let(:past_notification) { create(:dossier_notification, :for_groupe_instructeur) }
+    let(:future_notification) { create(:dossier_notification, :for_groupe_instructeur, display_at: 1.day.from_now) }
+
+    it 'includes notifications where display_at is in the past or now' do
+      expect(DossierNotification.to_display).to include(past_notification)
+    end
+
+    it 'excludes notifications where display_at is in the future' do
+      expect(DossierNotification.to_display).not_to include(future_notification)
+    end
+  end
+
+  describe '.notifications_for' do
+    let!(:instructeur) { create(:instructeur) }
+    let!(:groupe_instructeur) { create(:groupe_instructeur, instructeurs: [instructeur]) }
+    let!(:other_groupe_instructeur) { create(:groupe_instructeur, instructeurs: [instructeur]) }
+    let!(:dossier) { create(:dossier, :accepte, groupe_instructeur:) }
+    let!(:other_dossier) { create(:dossier, :en_construction, groupe_instructeur: other_groupe_instructeur) }
+    let!(:notification_instructeur) { create(:dossier_notification, :for_instructeur, dossier:, instructeur:) }
+    let!(:notification_grp_instructeur) { create(:dossier_notification, :for_groupe_instructeur, dossier:, groupe_instructeur:) }
+    let!(:other_notification_instructeur) { create(:dossier_notification, :for_instructeur, dossier: other_dossier, instructeur:) }
+    let!(:other_notification_grp_instructeur) { create(:dossier_notification, :for_groupe_instructeur, dossier: other_dossier, groupe_instructeur: other_groupe_instructeur) }
+
+    context 'a given instructeur and one dossier' do
+      subject { DossierNotification.notifications_for_instructeur_dossier(instructeur, dossier) }
+
+      it { is_expected.to include(notification_instructeur) }
+      it { is_expected.to include(notification_grp_instructeur) }
+      it { is_expected.not_to include(other_notification_instructeur) }
+      it { is_expected.not_to include(other_notification_grp_instructeur) }
```

Comment:
idéalement pour que optimiser les tests, on essaie de regrouper les assertions qui utilisent le même setup dans le même `it` quand c'est possible, car la phase de setup est longue (création des dossiers, démarches, admins etc…).